reducer/src/test/java/com/graphicsfuzz/reducer/reductionopportunities/LiteralToUniformReductionOpportunitiesTest.java

```diff
@@ -25,40 +26,152 @@
 import com.graphicsfuzz.common.util.PipelineInfo;
 import com.graphicsfuzz.common.util.RandomWrapper;
 import com.graphicsfuzz.common.util.ShaderKind;
+import com.graphicsfuzz.util.Constants;
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Optional;
-import org.junit.Ignore;
 import org.junit.Test;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
 
 public class LiteralToUniformReductionOpportunitiesTest {
 
-  @Test @Ignore
+  @Test
   public void testReplaceSimpleInt() throws Exception {
-    final String shader = "void main() { int a = 1; }";
 
-    // TODO: fix this.
-    final String shaderReplaced = " ...uniform decl... void main() { int a = ...; }";
+    final String vertexShader = "void main() { "
+        + "int a = 1;}";
 
+    final String vertexShaderReplaced = "uniform int _GLF_uniform_int_values[2];"
+        + "void main()"
+        + "{"
+        + "  int a = _GLF_uniform_int_values[0];"
+        + "}";
+
+    final String fragmentShader = "void main() { "
+        + "int a = 2;}";
+
+    final String fragmentShaderReplaced = "uniform int _GLF_uniform_int_values[2];"
+        + "void main()"
+        + "{"
+        + "  int a = _GLF_uniform_int_values[1];"
+        + "}";
+
+    final List<TranslationUnit> shaders = new ArrayList<>();
+    shaders.add(ParseHelper.parse(vertexShader, ShaderKind.VERTEX));
+    shaders.add(ParseHelper.parse(fragmentShader, ShaderKind.FRAGMENT));
     final PipelineInfo pipelineInfo = new PipelineInfo();
     final ShaderJob shaderJob = new GlslShaderJob(Optional.empty(),
-        pipelineInfo, ParseHelper.parse(shader, ShaderKind.FRAGMENT));
+        pipelineInfo, shaders);
```

Comment:
Here, add:

```
assertEquals(0, pipelineInfo.getNumUniforms());
```

I believe this should pass at present.

However ... (see next comment)